cli/oyster-cvm/src/commands/build.rs

```diff
@@ -0,0 +1,45 @@
+use crate::types::Platform;
+use anyhow::{Context, Result};
+use log::info;
+use std::process::Command;
+
+pub fn build_enclave_image(
+    platform: Platform,
+    docker_compose: &str,
+    docker_images: &[String],
+    output: &str,
+) -> Result<()> {
+    info!("Building enclave image with:");
+    info!("  Platform: {}", platform.as_str());
+    info!("  Docker compose: {}", docker_compose);
+    info!("  Docker images: {}", docker_images.join(", "));
+
+    let docker_images_list = docker_images.join(" ");
+
+    let nix_expr = format!(
+        r#"((builtins.getFlake "github:marlinprotocol/oyster-monorepo").packages.{}.sdks.docker-enclave.override {{compose={};dockerImages=[{}];}}).default"#,
+        platform.nix_arch(),
+        docker_compose,
+        docker_images_list
+    );
+
+    let status = Command::new("nix")
+        .args([
+            "build",
+            "--impure",
```

Comment:
why impure?

Reply:
since the enclave derivation needs kernel (external) access, --impure is needed to let nix allow that.

Reply:
then this won't be the reproducible build. Isn't that required?

Reply:
I think this is a must have to run the enclave derivation, and shouldn't be a problem to run in most systems. @roshanrags Please lemme know if I'm missing anything here.

Reply:
can you point to the derivation which needs impure? i am looking at this file https://github.com/marlinprotocol/oyster-monorepo/blob/master/sdks/docker-enclave/default.nix .

Reply:
`compose ? ./. + "/docker-compose.yml"`
since we are passing docker-compose and docker-images from the user's system as args, nix needs to resolve their paths which are flagged as external source.

Reply:
these are just files, copying the file shouldn't require impure. Let's take discussion offline.

Reply:
With flakes, all dependencies are expected to be defined in the flake for pure builds. Here we are using the docker related files that flake isn't aware of, hence using impure allowed inclusion of those files in the build.